quinn-proto/src/endpoint.rs

```diff
@@ -901,9 +979,21 @@ struct ConnectionIndex {
 }
 
 impl ConnectionIndex {
+    /// Associate an incoming connection with its initial destination CID
+    fn insert_initial_incoming(&mut self, dst_cid: ConnectionId, incoming_key: usize) {
+        self.connection_ids_initial
+            .insert(dst_cid, RouteDatagramTo::Incoming(incoming_key));
+    }
+
+    /// Remove an association between an incoming connection and its initial destination CID
+    fn remove_initial_incoming(&mut self, dst_cid: ConnectionId) {
```

Comment:
nit: the behavior here doesn't seem to be `Incoming`-specific
```suggestion
    fn remove_initial(&mut self, dst_cid: ConnectionId) {
```

Reply:
Done.